invalid memory address or nil pointer dereference, after SIGINT

**Describe the bug**

I ran `git add -p` and interrupted that with Ctrl+C, got this panic:

```
$ git add -p

Exiting because of "interrupt" signal.
panic: runtime error: invalid memory address or nil pointer dereference
[signal 0xc0000005 code=0x1 addr=0x68 pc=0x5f07d9]

goroutine 7 [running]:
github.com/git-lfs/git-lfs/config.(*Configuration).Cleanup(0x0, 0x0, 0x0)
        D:/a/git-lfs/git-lfs/config/config.go:436 +0x49
github.com/git-lfs/git-lfs/commands.Cleanup()
        D:/a/git-lfs/git-lfs/commands/commands.go:274 +0x38
sync.(*Once).doSlow(0xc000026fe0, 0xa3f400)
        go/src/sync/once.go:66 +0xf3
sync.(*Once).Do(...)
        go/src/sync/once.go:57
main.main.func1(0xc0000242a0, 0xc000026fe0)
        D:/a/git-lfs/git-lfs/git-lfs.go:22 +0x14c
created by main.main
        D:/a/git-lfs/git-lfs/git-lfs.go:19 +0xee
```

**To Reproduce**

Unknown, but the working tree did contain several modified files tracked by Git LFS.

**Expected behavior**

The process should terminate without panic.

**System environment**

Microsoft Windows Version 20H2 (OS Build 19042.867)

Windows Terminal Preview 1.7.572.0 with a profile that runs `"%ProgramFiles%\Git\bin\bash.exe" -i -l`

**Output of `git lfs env`**

```
git-lfs/2.13.2 (GitHub; windows amd64; go 1.14.13; git fc664697)
git version 2.30.2.windows.1

Endpoint=https://bitbucket.REDACTED.REDACTED.REDACTED/REDACTED/REDACTED.git/info/lfs (auth=none)
  SSH=git@bitbucket.REDACTED.REDACTED.REDACTED:REDACTED/REDACTED.git
Endpoint (jenkins)=file://///jenkins-srv.REDACTED.REDACTED.REDACTED/Jenkins/git-reference/REDACTED/REDACTED.git (auth=none)
LocalWorkingDir=C:\Projects\REDACTED\REDACTED\fix
LocalGitDir=C:\Projects\REDACTED\REDACTED\REDACTED.git\worktrees\fix
LocalGitStorageDir=C:\Projects\REDACTED\REDACTED\REDACTED.git
LocalMediaDir=C:\Projects\REDACTED\REDACTED\REDACTED.git\lfs\objects
LocalReferenceDirs=
TempDir=C:\Projects\REDACTED\REDACTED\REDACTED.git\lfs\tmp
ConcurrentTransfers=8
TusTransfers=false
BasicTransfersOnly=false
SkipDownloadErrors=false
FetchRecentAlways=false
FetchRecentRefsDays=7
FetchRecentCommitsDays=0
FetchRecentRefsIncludeRemotes=true
PruneOffsetDays=3
PruneVerifyRemoteAlways=true
PruneRemoteName=origin
LfsStorageDir=C:\Projects\REDACTED\REDACTED\REDACTED.git\lfs
AccessDownload=none
AccessUpload=none
DownloadTransfers=basic,lfs-standalone-file
UploadTransfers=basic,lfs-standalone-file
GIT_EXEC_PATH=C:/Program Files/Git/mingw64/libexec/git-core
GIT_SSH=C:\Program Files\PuTTY\plink.exe
GIT_SSH_VARIANT=putty
git config filter.lfs.process = "git-lfs filter-process"
git config filter.lfs.smudge = "git-lfs smudge -- %f"
git config filter.lfs.clean = "git-lfs clean -- %f"
```

**Additional context**

None

Hey, thanks for the report, and my apologies for the delayed response.

From the Go stack trace it looks like there's some contention going on as the Git LFS process shuts down.  The specific location of the panic appears to be from trying to [acquire](https://github.com/git-lfs/git-lfs/blob/dc68f8afff173ca022c5cc55b3839dbcb1dab944/config/config.go#L436) a mutex while inside the cleanup [call](https://github.com/git-lfs/git-lfs/blob/dc68f8afff173ca022c5cc55b3839dbcb1dab944/git-lfs.go#L22).

One intriguing detail is that the `"Exiting because of ... signal."` message is printed first in your trace, but [appears after](https://github.com/git-lfs/git-lfs/blob/dc68f8afff173ca022c5cc55b3839dbcb1dab944/git-lfs.go#L23) the cleanup call in the code.  On first blush I'm not entirely sure what to make of that: while I can reproduce the panic with a `SIGINT`, I don't get that initial message.  Compiler optimization reordering code?  Another goroutine running the same code?  Another process getting the same signal and exiting cleanly?  Perhaps the latter is most likely?

Regardless of that detail, I can reproduce an otherwise similar panic with a Ctrl-C on macOS; in this case:
```
panic: runtime error: invalid memory address or nil pointer dereference
[signal SIGSEGV: segmentation violation code=0x1 addr=0x68 pc=0x11ac2a2]

goroutine 35 [running]:
github.com/git-lfs/git-lfs/config.(*Configuration).Cleanup(0x0, 0x0, 0x0)
        work/chrisd8088/git-lfs/config/config.go:436 +0x42
...
```

I'll try to dig a bit more into what's going on here.  Thanks again for the report!

--- a/config/config.go
+++ b/config/config.go
@@ -433,6 +433,9 @@ func (c *Configuration) Filesystem() *fs.Filesystem {
 }
 
 func (c *Configuration) Cleanup() error {
+	if c == nil {
+		return nil
+	}
 	c.loading.Lock()
 	defer c.loading.Unlock()
 	return c.fs.Cleanup()

